### PR TITLE
Feature/advanced admin filters

### PR DESCRIPTION
**Pull request by appsol**
### Summary
https://trello.com/c/Zl6sXM8H

- Added additional filter options to Social Media filter
- Added additional filter options to Phone filter

### Development checklist
- [x] Changes have been made to the API?
  - [x] If so, the OpenAPI specification has been updated
- [ ] Database migrations have been added?
  - [ ] If so, the MySQL Workbench ERD has been updated
- [x] The code has been linted `./develop composer fix:style`

### Release checklist
NA

### Notes
NA

### CODE REVIEW

File: app/Http/Filters/Organisation/HasSocialMediasFilter.php [hunk 1]
```diff
@@ -15,10 +18,28 @@ class HasSocialMediasFilter implements Filter
...
+            case SocialMedia::TYPE_FACEBOOK:
+            case SocialMedia::TYPE_TWITTER:
+            case SocialMedia::TYPE_INSTAGRAM:
+            case SocialMedia::TYPE_YOUTUBE:
+            case SocialMedia::TYPE_OTHER:
+                $query = $query->whereExists(function ($query) use ($value) {
+                    $query->select(DB::raw(1))
+                        ->from('social_medias')
+                        ->whereRaw('organisations.id = social_medias.sociable_id')
+                        ->whereRaw('social_medias.sociable_type = ?', [table(Organisation::class)])
```
Comment (matthew-inamdar) [inline app/Http/Filters/Organisation/HasSocialMediasFilter.php:37]: The morph map overrides the default:
```suggestion
                        ->whereRaw('social_medias.sociable_type = ?', ['organisations'])
```

https://github.com/Healthy-London-Partnership/api/blob/bef6b85ed66c4c0b1cbedb3829f7c15848d735ff/app/Providers/ModelServiceProvider.php#L56-L61

**Review by matthew-inamdar — CHANGES_REQUESTED** [review]
Looks good. Left a comment to take a look at.

File: app/Http/Filters/Organisation/HasSocialMediasFilter.php (hunk 1 above)
Comment (appsol) [reply app/Http/Filters/Organisation/HasSocialMediasFilter.php]: Done.
Sorry I assumed the table function was designed to work with the morph map. Having had a look inside I can see it just calculates a standard table name. I'll take this into account in future.

Comment (matthew-inamdar) [reply app/Http/Filters/Organisation/HasSocialMediasFilter.php]: @appsol Yeah, it does. Sorry I completely missed the use of `table()` - that would have worked. I'd still avoid the use in the case though, as morph map just happens to be using the same name as the tables, but there's nothing stopping that from changing in the future.

**Review by matthew-inamdar — APPROVED** [review]
LGTM 👍

**Pull request merged**